### PR TITLE
[kie-issues#969]: Negated value in accumulate function ends with Invalid expression error

### LINKED ISSUE #969: Negated value in accumulate function ends with Invalid expression error
When negating a value in a DRL rule, in an accumulate function, like this: 
```
accumulate ($p: Person (getName().startsWith("M")); 
                     $sum : sum(-$p.getAge()))
```
The build of the executable model ends with error: 
`Invalid expression -$p.getAge()`

Test case that could be added to org.drools.model.codegen.execmodel.AccumulateTest class: 
https://gist.github.com/baldimir/dd594f0b6e789a61284e134b7ee8f529

### PR DESCRIPTION
**Pull request by yesamer**
Closes: https://github.com/apache/incubator-kie-issues/issues/969

```
accumulate ($p: Person (getName().startsWith("M")); 
                     $sum : sum(-$p.getAge()))
```
This case leads to an `Invalid expression -$p.getAge()` error.

That is because the UnaryExpression types are not managed in the `AccumulateVisitor` logic.
To fix the issue, I managed that case relying on the existing BinaryExpression logic management, which fits well for the `UnaryExpression` case according to my analysis. 
Please correct me if I'm wrong @lucamolteni @mariofusco.

In addition, we added a check to ensure that the accumulate function always has no more than one parameter (kudos to @gitgabrio ). I added that check for the executable Model case only

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Review by lucamolteni — APPROVED** [review]

File: drools-model/drools-model-codegen/src/test/java/org/drools/model/codegen/execmodel/AccumulateTest.java [hunk 1]
```diff
@@ -4204,4 +4206,80 @@ public void testJoinInAccumulate() {
...
+                "  accumulate ( $i : Integer() and $p: Person ( name.length >= $i ); \n" +
+                "                $sum : sum($p.getAge(), $p.getName())  \n" +
+                "              )                          \n" +
+                "then\n" +
+                "  insert(new Result($sum));\n" +
+                "end";
+
+        Results results = createKieBuilder(str).getResults();
+        if (testRunType.isExecutableModel()) {
+            System.out.println(results.getMessages(Message.Level.ERROR).get(0).getText());
```
Comment (gitgabrio) [inline drools-model/drools-model-codegen/src/test/java/org/drools/model/codegen/execmodel/AccumulateTest.java]: Thanks @yesamer 
Could you please remove this System.out.println ?

File: drools-model/drools-model-codegen/src/main/java/org/drools/model/codegen/execmodel/generator/visitor/accumulate/AccumulateVisitor.java [hunk 2]
```diff
@@ -406,10 +414,10 @@ private boolean parameterNeedsConvertionToMethodCallExpr(Expression accumulateFu
         return accumulateFunctionParameter.isMethodCallExpr() || accumulateFunctionParameter.isArrayAccessExpr() || accumulateFunctionParameter.isFieldAccessExpr();
     }
 
-    private Optional<NewBinding> binaryExprParameter(PatternDescr basePattern, AccumulateDescr.AccumulateFunctionCallDescr function, MethodCallExpr functionDSL, String bindingId, String accumulateFunctionParameterStr) {
+    private Optional<NewBinding> binaryOrUnaryExprParameter(PatternDescr basePattern, AccumulateDescr.AccumulateFunctionCallDescr function, MethodCallExpr functionDSL, String bindingId, String accumulateFunctionParameterStr) {
```
Comment (gitgabrio) [inline drools-model/drools-model-codegen/src/main/java/org/drools/model/codegen/execmodel/generator/visitor/accumulate/AccumulateVisitor.java]: HI @yesamer 
Would it make sense to rename it as `bindingParameter` or similar ?
1. if, by chance, we discover that another kind of expression has to be managed here, the method name would become awfull
2. usually, the method name express the returned object
3. IINW, nothing prevent this method to be invoked by somehting different then a `BinaryExpr` or a `UnaryExpr` (if it is implicitily so, I think there should be a check/guard somehow)

WDYT ?

**Review by gitgabrio — CHANGES_REQUESTED** [review]
HI @yesamer 
There is a System.out.println to remove. And maybe a small refactoring/enforcement

File: drools-model/drools-model-codegen/src/main/java/org/drools/model/codegen/execmodel/generator/visitor/accumulate/AccumulateVisitor.java (hunk 2 above)
Comment (yesamer) [reply drools-model/drools-model-codegen/src/main/java/org/drools/model/codegen/execmodel/generator/visitor/accumulate/AccumulateVisitor.java]: Thank you, done!

File: drools-model/drools-model-codegen/src/test/java/org/drools/model/codegen/execmodel/AccumulateTest.java (hunk 1 above)
Comment (yesamer) [reply drools-model/drools-model-codegen/src/test/java/org/drools/model/codegen/execmodel/AccumulateTest.java]: Done

**Review by gitgabrio — APPROVED** [review]
Thanks @yesamer LGTM!

**Review by baldimir — APPROVED** [review]

**Pull request merged**